### DISCUSSION TITLE
Cross repository code snippets

### DISCUSSION THREAD

**Discussion by thorntonryan** [community/community#10412] [Projects and Issues]
### Suggested Behavior

I think it'd be great if issues supported cross-repository code snippets.

My team often works between several related repositories (e.g. primary application and a supporting library).

Quite often, while working in Repository A, we need to make reference to a specific piece of code from Repository B.

GitHub doesn't support this, so we end up with the very manual pattern of:

``````
[permalink](https://github.deere.com/ORG/REPO/blob/32ffAA3be40f5b813f90c4799d8f7aad35428c60/Foo/bar.cpp#L820)

// copy/paste code we want to talk about
```c++
void bar() {
    ...
}
```
``````

I think it'd be great if GitHub supported this automagically.

Thanks!
Ryan

### Current Behavior

Currently, code snippets are restricted to [permalinks](https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-a-permanent-link-to-a-code-snippet), but only within the same repository.

**Comment by kriegaex** [comment]
I second this request. To me, it is absolutely essential. See also [my personal support ticket](https://support.github.com/ticket/personal/0/1512413). Quoting from there:

> I actually really wanted to inline a snippet from another repository. (...) This is not very nice and intuitive. I would strongly encourage you to permit inlining snippets from other repositories, too, at least with a special syntax, maybe something like `?inline=true` at the end or a special markdown syntax for it. Inlining snippets when discussing changes or bugs is very helpful. And as you know, many organisations or users have multiple related repositories. Sometimes, problems are also between two independent projects or forks of the same one.

**Comment by gian1200** [comment]
Was there any update regarding this feature request on another location? It seems to still be a missing functionality

**Reply by kasir-barati** [reply]
I think we've been in limbo for quite sometime. No news as in this feedback was discarded, or as in it has been accepted but have not gotten around it yet :thinking:. Lack of communication sometimes is really frustrating.

**Comment by edwardzjl** [comment]
Any update? This is really useful.

**Comment by sawatts** [comment]
Most of our work is factored into multiple repos, and cross-reference each other for base implementations etc.  I don't understand the rational behind this restriction.

If its a load issue, then at least let it work with in the same org - but that too is an arbitrary restriction.

**Comment by jfullstackdev** [comment]
there is I guess one that might answer that, plus it's the environment of the AI, the MCP of GitHub itself, where your AI sees all of the repos that you have access, so it can reference even PR or Issues in your Org

in VSCode Extensions, you will see this in MCP part

<img width="537" height="46" alt="image" src="https://github.com/user-attachments/assets/fb5d3380-2c23-482e-b5e6-95730ec08d72" />

then click the Globe icon, 

it will bring you here and simply install GitHub or any other that you want,

<img width="855" height="684" alt="image" src="https://github.com/user-attachments/assets/215ef6cc-c280-4a3b-87bc-a67a699eef31" />

MCPs are designed to add context to AI without pasting it, so it can reference code snippets in any of your repo, just tell it
